### DISCUSSION TITLE
how to make open-webui secure on a local network

### DISCUSSION THREAD

**Discussion by iplayfast** [open-webui/open-webui#4020] [General]
**Is your feature request related to a problem? Please describe.**
I have open-webui running on my main computer, and I am accessing it from my lap top. IE http://192.168.1.15:3000
Because it's http and not https I can't use the microphone or camera on my laptop. 
**Describe the solution you'd like**
How do I set it up so that it's https?

**Comment by thiswillbeyourgithub** [comment]
Actually it is possible to add an exception to any URL but it's well hidden. I made a discussion post there but pinging didn't work so [linking here instead](https://github.com/open-webui/open-webui/discussions/4352)

This way I can use voice recognition / STT on all my devices including android

**Comment by GrayXu** [comment]
You just need to set up Caddy or Nginx in front.

**Reply by thiswillbeyourgithub** [reply]
Do you mind sharing your docker compose ? I was not able to do make it work last time I tried.

**Comment by DeCodeTheWeb** [comment]
I've been looking for a solution to this, but I don't want to expose my server to the internet (as with nginx).  I only want to be able to use it locally (from a different device).  Currently it works fine with http but not https which means I can't use the mic as the OP notes.  Is there really no solution to this but expose the entire server to the world wide web with Nginx?
